lib/charms/hpc_libs/v0/slurm_ops.py

```diff
@@ -658,6 +655,22 @@ def _install_service(self) -> None:
         except (apt.PackageNotFoundError, apt.PackageError) as e:
             raise SlurmOpsError(f"failed to install {self._service_name}. reason: {e}")
 
+    def _create_state_save_location(self) -> None:
+        """Create `StateSaveLocation` for Slurm services.
+
+        Notes:
+            `StateSaveLocation` is used by slurmctld, slurmd, and slurmdbd
+            to checkpoint runtime information should a service crash, and it
+            serves as the location where the JWT token used to generate user
+            access tokens is stored as well.
+        """
+        _logger.debug("creating slurm `StateSaveLocation` directory")
+        target = self.var_lib_path / "checkpoint"
+        target.mkdir(mode=0o755, parents=True, exist_ok=True)
+        self.var_lib_path.chmod(0o755)
+        shutil.chown(self.var_lib_path, "slurm", "slurm")
+        shutil.chown(target, "slurm", "slurm")
```

Comment:
Nit: Possibly changing the permissions for `var_lib_path` should be outside `_create_state_save_location`? Seems unrelated to creating the directory itself.

Reply:
`self.var_lib_path.chmod(0o755)` is used here because of the perturbations between each of the Slurm daemon packages. `/var/lib/slurm` is created with `0o755` file permissions for `slurmctld`, `slurmd`, and `slurmdbd`, but not `slurmrestd`.

`parents=True` will ensure that `/var/lib/slurm` is created on `slurmrestd` machines, but the file permissions mode `0o755` will only be applied to the `StateSaveLocation` directory `/var/lib/slurm/checkpoint`, not `/var/lib/slurm`. Its mode will be `0o775` which is looser than I would like, so that's why `self.var_lib_path.chmod(0o755)` is there. It's more a stop-gap to ensure `/var/lib/slurm` looks and feels the same on all the Slurm nodes. 

If we end up using `var_lib_path` in more places than just in `_AptManager` for creating the `StateSaveLocation`, and `_JWTKeyManager` for storing the JWT key in `StateSaveLocation`, then I'd agree that we should move managing the permissions for `var_lib_path` outside of `_create_state_save_location`.